PACKAGE = "l3cam_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Parameters
# gen.add(name, paramtype, level, description, default, min, max, edit_method)

# CHANGE_POLARIMETRIC_CAMERA_BRIGHTNESS
gen.add("change_polarimetric_camera_brightness", int_t, 0, "Change polarimetric camera brightness", 127, 0, 255)
# CHANGE_POLARIMETRIC_CAMERA_BLACK_LEVEL
gen.add("change_polarimetric_camera_black_level", double_t, 1, "Change polarimetric camera black level", 6.0, 0.0, 12.5)
# ENABLE_POLARIMETRIC_CAMERA_AUTO_GAIN
gen.add("enable_polarimetric_camera_auto_gain", bool_t, 2, "Enable polarimetric camera auto egain", True)
# CHANGE_POLARIMETRIC_CAMERA_AUTO_GAIN_RANGE
gen.add("change_polarimetric_camera_auto_gain_range_minimum", double_t, 3, "Change polarimetric camera auto gain time range minimum", 0.0, 0.0, 48.0)
gen.add("change_polarimetric_camera_auto_gain_range_maximum", double_t, 4, "Change polarimetric camera auto gain time range maximum", 48.0, 0.0, 48.0)
# CHANGE_POLARIMETRIC_CAMERA_GAIN
gen.add("change_polarimetric_camera_gain", double_t, 5, "Change polarimetric camera gain", 24.0, 0.0, 48.0)
# ENABLE_POLARIMETRIC_CAMERA_AUTO_EXPOSURE_TIME
gen.add("enable_polarimetric_camera_auto_exposure_time", bool_t, 6, "Enable polarimetric camera auto exposure time", True)
# CHANGE_POLARIMETRIC_CAMERA_AUTO_EXPOSURE_TIME_RANGE
gen.add("change_polarimetric_camera_auto_exposure_time_range_minimum", double_t, 7, "Change polarimetric camera auto exposure time range minimum", 33.456, 33.456, 1000000.0)
gen.add("change_polarimetric_camera_auto_exposure_time_range_maximum", double_t, 8, "Change polarimetric camera auto exposure time range maximum", 1000000.0, 33.456, 1000000.0)
# CHANGE_POLARIMETRIC_CAMERA_EXPOSURE_TIME
gen.add("change_polarimetric_camera_exposure_time", double_t, 9, "Change polarimetric camera exposure time (us)", 500000.0, 33.456, 1000000.0)

exit(gen.generate(PACKAGE, "l3cam_ros", "PolarimetricCamera"))
